Content.Client/_NF/Shuttles/UI/ShuttleNavControl.xaml.cs

```diff
@@ -127,5 +164,164 @@ private void NfDrawBlips(DrawingHandleBase handle, List<BlipData> blipDataList)
+            var coords = _coordinates.Value.Offset(relativeWorldPos);
+            OnRadarClick?.Invoke(coords);
+        }
+
+        private void DrawBlipShape(DrawingHandleScreen handle, Vector2 position, float size, Color color, RadarBlipShape shape)
+        {
+            switch (shape)
+            {
+                case RadarBlipShape.Circle:
+                    handle.DrawCircle(position, size, color);
+                    break;
+                case RadarBlipShape.Square:
+                    var halfSize = size / 2;
+                    var rect = new UIBox2(
+                        position.X - halfSize,
+                        position.Y - halfSize,
+                        position.X + halfSize,
+                        position.Y + halfSize
+                    );
+                    handle.DrawRect(rect, color);
+                    break;
+                case RadarBlipShape.Triangle:
+                    var points = new Vector2[]
+                    {
+                    position + new Vector2(0, -size),
+                    position + new Vector2(-size * 0.866f, size * 0.5f),
+                    position + new Vector2(size * 0.866f, size * 0.5f)
+                    };
+                    handle.DrawPrimitives(DrawPrimitiveTopology.TriangleList, points, color);
+                    break;
+                case RadarBlipShape.Star:
+                    DrawStar(handle, position, size, color);
+                    break;
+                case RadarBlipShape.Diamond:
+                    var diamondPoints = new Vector2[]
+                    {
+                    position + new Vector2(0, -size),
+                    position + new Vector2(size, 0),
+                    position + new Vector2(0, size),
+                    position + new Vector2(-size, 0)
+                    };
+                    handle.DrawPrimitives(DrawPrimitiveTopology.TriangleFan, diamondPoints, color);
+                    break;
+                case RadarBlipShape.Hexagon:
+                    DrawHexagon(handle, position, size, color);
+                    break;
+                case RadarBlipShape.Arrow:
+                    DrawArrow(handle, position, size, color);
+                    break;
+            }
+        }
+
+        private void DrawStar(DrawingHandleScreen handle, Vector2 position, float size, Color color)
+        {
+            const int points = 5;
+            const float innerRatio = 0.4f;
+            var vertices = new Vector2[points * 2];
+
+            for (var i = 0; i < points * 2; i++)
+            {
+                var angle = i * Math.PI / points;
+                var radius = i % 2 == 0 ? size : size * innerRatio;
+                vertices[i] = position + new Vector2(
+                    (float)Math.Sin(angle) * radius,
+                    -(float)Math.Cos(angle) * radius
+                );
+            }
+
+            handle.DrawPrimitives(DrawPrimitiveTopology.TriangleFan, vertices, color);
```

Comment:
```suggestion
            const int points = 5;
            const float innerRatio = 0.4f;
            var vertices = new Vector2[points * 2 + 1];

            vertices[0] = position;
            for (var i = 0; i < points * 2; i++)
            {
                var angle = i * Math.PI / points;
                var radius = i % 2 == 0 ? size : size * innerRatio;
                vertices[i + 1] = position + new Vector2(
                    (float)Math.Sin(angle) * radius,
                    -(float)Math.Cos(angle) * radius
                );
            }

            handle.DrawPrimitives(DrawPrimitiveTopology.TriangleFan, vertices, color);
```
If this is a triangle fan, there should be a point in the center first.
Be warned, haven't tested this.